overlord/snapstate/snapstate.go

```diff
@@ -168,6 +168,23 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 	addTask(setupAliases)
 	prev = setupAliases
 
+	// run refresh, revert or install hook
+	if snapst.HasCurrent() {
+		if snapsup.Flags.Revert {
+			revertHook := SetupRevertHook(st, snapsup.Name())
```

Comment:
I was pondering that too. This may be sensible thing to do, will run this by @niemeyer.